explainer_comparison/Explainer.py

```diff
@@ -5,18 +5,21 @@
 # ------------------------------------------------------------------------------------------------------
 from abc import ABC, abstractmethod
 import pandas as pd
+from explainer_comparison.constants import MODE
 
 
 class Explainer(ABC):
     def __init__(self,
                  model,
                  X_train: pd.DataFrame,
                  y_train: pd.DataFrame,
-                 y_pred: pd.DataFrame = None):
+                 y_pred: pd.DataFrame = None,
```

Comment:
y_train and y_pred could also be pd.series or np.array. please use Union[pd.DataFrame, pd.Series, np.ndarray] and perform an internal check and treat accordingly

Reply:
in __init__ updated as: y_train: Union[pd.DataFrame, pd.Series, np.ndarray],
                 y_pred: Union[pd.DataFrame, pd.Series, np.ndarray, None] = None
+added function def _check_input(self, data)